data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java

```diff
@@ -62,23 +64,30 @@ public AadAuthenticationHelper(@NotNull ConnectionStringBuilder csb) throws URIS
     }
 
     String acquireAccessToken() throws DataServiceException  {
+        if(lastAuthenticationResult != null && lastAuthenticationResult.getExpiresOnDate().after(new Date())){
+            return lastAuthenticationResult.getAccessToken();
+        }
+
         try {
             switch (authenticationType) {
                 case AAD_APPLICATION_KEY:
-                    return acquireAadApplicationAccessToken().getAccessToken();
+                    lastAuthenticationResult = acquireAadApplicationAccessToken();
+                    return lastAuthenticationResult.getAccessToken();
```

Comment:
Consider combing all return statements in the switch block and have just one after it